docs/azure-openshift.md

```diff
@@ -0,0 +1,162 @@
+# Use external dns operator on Openshift in Azure environment
+## Note: These instructions are not for Azure private DNS. 
+
+### Steps
+
+1. Export your kubeconfig
+```bash
+ $ export KUBECONFIG=~/<path to>/kubeconfig
+```
+
+2. Check user. The user shall access to kube-system.
+
+```bash
+ $ oc whoami
+ system:admin
+```
+
+3. Fetch the values from azure-credentials secret present in kube-system
+```bash
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_client_id}} | base64 -d
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_client_secret}} | base64 -d
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_region}} | base64 -d
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_resource_prefix}} | base64 -d
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_resourcegroup}} | base64 -d
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_subscription_id}} | base64 -d
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_tenant_id}} | base64 -d
+
+```
+
+4. Login to azure with base64 decoded values you get from above
+
+```bash
+$ az login --service-principal -u <azure_client_id> -p <azure_client_secret> --tenant <azure_tenant_id>
+```
+
+5. Get the routes to check the domain
+```bash
+oc get routes --all-namespaces | grep console
+openshift-console          console             console-openshift-console.apps.misalunk-azure.qe.azure.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
+openshift-console          downloads           downloads-openshift-console.apps.misalunk-azure.qe.azure.devcluster.openshift.com                     downloads           http    edge/Redirect          None
+```
+
+6. Get the dns zone list w.r.t your azure_resourcegroup. Get it from the base64 decoded output.
+```bash
+$ az network dns zone list --resource-group <azure_resourcegroup>
+[]
+```
+Initially you will see nothing. So you will have create a zone for your resourcegroup.
+
+7. Create a dns zone for you resourcegroup.
+```bash
+ az network dns zone create -g <azure_resourcegroup> -n <pick up the zone name from the route  eg- the part after apps. eg - misalunk-azure.qe.azure.devcluster.openshift.com>
+```
+
+8. Then create CR as follows -
+
+```yaml
+  apiVersion: externaldns.olm.openshift.io/v1alpha1
+  kind: ExternalDNS
+  metadata:
+    name: sample-azure
+  spec:
+    domains:
+    - filterType: Include
+      matchType: Exact
+      name: misalunk-azure.qe.azure.devcluster.openshift.com
+    provider:
+      type: Azure
+    source:
+      hostnameAnnotation: Allow
+      openshiftRouteOptions:
+        routerName: default
+      type: OpenShiftRoute
```

Comment:
before I can see the section below for publishing service to external DNS, is that still supported or removed? 
```
  source:
    type: Service
    service:
      serviceType:
        - LoadBalancer
        - ClusterIP
    annotationFilter:
      external-dns.mydomain.org/publish: "yes"
    fqdnTemplate:
      - "{{.Name}}.test-azure.example.com"
```

Reply:
These are not needed.

Reply:
BTW I have sent PRs for sample CR's. @alebedev87 has to approve them yet though.